.github/workflows/build-docker-images.yml

```diff
@@ -0,0 +1,67 @@
+name: Build and Push Docker Images
+
+on:
+  push:
+    branches: [develop, 'GEN*', 'gen*']
+    paths:
+      - 'scripts/references/**'
+      - '.github/workflows/build-docker-images.yml'
+  workflow_dispatch:
+
+jobs:
+  build_references_docker:
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: sage-bionetworks/genie-bpc-pipeline
```

Comment:
Just a quick note: did we want the different folders to be tags OR following the same structure as we have now in dockerhub with

...-pipeline-case-selection:develop

Reply:
i guess this is achieved in the Build and Push Docker Image step. ```tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:references-${{ github.ref_name }}```

Reply:
Right now the structure is `<registry>/<repo>:<folder_name>-<branch>` mainly because in GHCR, I don't believe we can do `<registry>/<repo>-<folder_name>:-<branch>`. I tried to concatenate to the `repo` part of the name but GHCR doesn't allow you to override the repo name? So I have it attached to the tag. 

But yes have a tag for each branch so we can do dev work and not break the current working image.